### ISSUE TITLE
Proposal for new doc translation instructions and process

### ISSUE THREAD

**Issue by frankbu** [istio/istio.io#8886]
The current instructions for adding new translations has issues. See https://github.com/istio/istio.io/pull/8205#issuecomment-700266489 for details.

Some ideas to address the issues have been discussed:

* Make copies of only docs that are localized into a new language
* When the user is viewing a documentation set in a non-English language, links within that documentation set should point to the localized version when available, and the English version (in the English URL path) when not.
* When we cut a release branch, all translated docs in the master branch become unavailable (not up to date) by default, until the translation team updates or otherwise marks the doc as up to date (i.e., still valid in the next release).

**Comment by jihoon-seo** [comment]
Hello @frankbu , I am Jihoon Seo, and I am currently:
- one of K8s sig-docs-ko-owners (approvers), and
- one of CNCF Glossary project approvers.

IMHO:

> Make copies of only docs that are localized into a new language

I agree. 
The current instructions for adding new translations instructs L10n contributor to create a full copy of the `content/en` directory for your language.
And according to [your comment](https://github.com/istio/istio.io/pull/8205#issuecomment-700266489), L10n contributor is required to not leave each file untranslated, which is almost impossible for the initial L10n team to achieve.

One possible additional option might be:
- Create a branch for L10n, let's say `dev-ko`.
- Initial Korean L10n team accumulates some translated files into `dev-ko` branch.
- When it gets okay to display "Korean" option in Istio website, then merge `dev-ko` branch into `master` branch.
  - cf. In CNCF Glossary project, when one language gets having more than or equal to 10 terms, then that language can be served in CNCF Glossary website.

> When the user is viewing a documentation set in a non-English language, links within that documentation set should point to the localized version when available, and the English version (in the English URL path) when not.

This would be great. Let me introduce K8s website case.
- In sidebar: Let's see https://kubernetes.io/ko/docs/concepts/policy/ for example.
![image](https://user-images.githubusercontent.com/46767780/160059610-9ae1b9b6-eeb9-4604-bc33-5889ea37677f.png)
- In content: It is handled manually; contributors have to set link to the translated ones if exist, and to the English ones if not.
  But some tools that help update exist:
  - [Tool 1]
    Issue: https://github.com/kubernetes/website/issues/31140
    PR: https://github.com/kubernetes/website/pull/31160
    Program: https://gist.github.com/riita10069/040509ff774a878becc8ab604cee9656
  - [Tool 2]
    Program: https://github.com/kubernetes/website/blob/main/scripts/linkchecker.py

Another example is [TypeScript-Website case](https://github.com/microsoft/TypeScript-Website/blob/v2/docs/How%20i18n%20Works%20For%20Site%20Copy.md).
Contributors can use `IntlLink` to achieve this.
> An IntlLink is a element, which is effectively an `<a>`, but that it knows the entire sitemap and the current locale.
> 
> This means the link will detect whether there is a version of the page available in the current locale if possible. You just write the English URL.

---

> When we cut a release branch, all translated docs in the master branch become unavailable (not up to date) by default, until the translation team updates or otherwise marks the doc as up to date (i.e., still valid in the next release).

Yes. But please understand that some delays in each L10n versions would be inevitable.

Each L10n team could adopt its own strategy to keep translated contents up to date.

For example:

K8s SIG Docs Korean L10n team has been keeping the Korean contents up to date using branching strategy like this:
  - (FYI: All the languages (en, ko, ...) are managed in one repo: https://github.com/kubernetes/website)
  - Every two weeks, the l10n teams of each language (e.g. Korean) create a new dev branch from `master`, like
    - `dev-1.20-ko.6` (Created on 2021-03-05 from `master`)
    - `dev-1.20-ko.7` (Created on 2021-03-20 from `master`)
  - Since `dev-1.20-ko.6` branch contains the Eng version of website source as of 2021-03-05,
  - and `dev-1.20-ko.7` branch contains the Eng version of website source as of 2021-03-20,
  - comparing `dev-1.20-ko.6` and `dev-1.20-ko.7` shows the changes in English version.
    - (The script that helps comparing: https://github.com/kubernetes/website/tree/main/scripts#diff_l10n_branchespy)

Meanwhile, K8s SIG Docs Chinese L10n team adopted another way.
- They find outdated Chinese-translated files, and update them.
- There is [a script](https://github.com/kubernetes/website/tree/main/scripts#lsyncsh) that checks if the English version of some localized contents have changed since a localized version has been committed.

In ReactJS website L10n project, a bot exists:
  - (FYI: The repo of the English version of the website: https://github.com/reactjs/reactjs.org)
  - (FYI: The repo of the Korean version of the website: https://github.com/reactjs/ko.reactjs.org)
  - There is a bot that
     - Every Monday,
     - This bot detects the changes in Eng repo,
     - and open a PR to Korean repo that suggests corresponding changes. ([Example](https://github.com/reactjs/ko.reactjs.org/pull/293))

PTAL!

**Comment by frankbu** [comment]
^^^ cc: @istio/wg-docs-maintainers-infra

**Comment by craigbox** [comment]
That all sounds wonderful, but we don't have any resources working on this.  I think we should put it on the backlog for docs work that we will recommend getting contractors to help with when we join the CNCF.

**Comment by zufardhiyaulhaq** [comment]
@craigbox did maintainers plan to use contractors to help translate the documentation? why not bring it to the community?

**Comment by craigbox** [comment]
No, maintainers planned to use contractors to help improve the documentation processes, so the community could more easily contribute translations. :)
